src/item_factory.cpp

```diff
@@ -721,59 +767,103 @@ void Item_factory::load( islot_software &slot, JsonObject &jo )
 
 void Item_factory::load( islot_ammo &slot, JsonObject &jo )
 {
-    slot.type = jo.get_string( "ammo_type" );
-    slot.casing = jo.get_string( "casing", slot.casing );
-    slot.damage = jo.get_int( "damage", 0 );
-    slot.pierce = jo.get_int( "pierce", 0 );
-    slot.range = jo.get_int( "range", 0 );
-    slot.dispersion = jo.get_int( "dispersion", 0 );
-    slot.recoil = jo.get_int( "recoil", 0 );
-    slot.def_charges = jo.get_long( "count" );
-    slot.ammo_effects = jo.get_tags( "effects" );
+    jo.assign( "ammo_type", slot.type );
+    jo.assign( "casing", slot.casing );
+    jo.assign( "damage", slot.damage );
+    jo.assign( "pierce", slot.pierce );
+    jo.assign( "range", slot.range );
+    jo.assign( "dispersion", slot.dispersion );
+    jo.assign( "recoil", slot.recoil );
+    jo.assign( "count", slot.def_charges );
+    jo.assign( "effects", slot.ammo_effects );
+    jo.assign( "loudness", slot.loudness );
+}
+
+void Item_factory::load_itype( JsonObject &jo )
+{
+    auto type = jo.get_string( "type" );
 
-    if( !jo.read( "loudness", slot.loudness ) ) {
-        slot.loudness = std::max( std::max( { slot.damage, slot.pierce, slot.range } ) * 3, slot.recoil / 3 );
+    itype *src = nullptr;
+    if( jo.has_string( "copy-from" ) ) {
```

Comment:
The main obstacle and reason for this decision was `it_tool` and `it_comest` which are causing other problems as identified by @BevapDin. It would seem reasonable to drop both of these derived classes and then go with a method at the start of each `load_*` call